### PR TITLE
clang: fix documentation

### PR DESCRIPTION
**Pull request by neheb**
Found with Wdocumentation

### CODE REVIEW

**Comment by neheb** [comment]
added extra somewhat related commit.

Even though this is C++17 now, should C++11 compatibility for the headers be kept?

**Comment by sbooth** [comment]
> Even though this is C++17 now, should C++11 compatibility for the headers be kept?

My personal opinion is that C++17 everywhere is fine, but I'm not sure what @scotchi and @ufleisch think.

**Comment by neheb** [comment]
For exiv2 for example, headers are C++11 compatible. Although [[nodiscard]] is used...

Maybe [[deprecated]] should be used instead. C++ compilers are required to ignore unknown attributes per the standard.

**Comment by sbooth** [comment]
I agree that `[[deprecated]]` makes more sense than the GNU version.

**Comment by neheb** [comment]
you want TAGLIB_DEPRECATED removal here?

**Comment by ufleisch** [comment]
Please do not re-add `TAGLIB_DEPRECATED`. I want TagLib 2.0 to build without any warnings. These functions were deprecated, but must remain there in order to keep TagLib 2.0 source-compatible with TagLib 1.x. The doxygen deprecation comments can still be there but must not have a corresponding deprecation attribute.

I also agree that C++17 everywhere is fine and that `[[deprecated]]` should be used instead of the GNU version.

**Comment by scotchi** [comment]
Just thinking outloud:  I wonder if it'd make sense to set a define via CMake that is something like TAGLIB_INTERN or something, and then to have the deprecation warning only be shown when these methods are used by external projects?

**Comment by ufleisch** [comment]
Concerning `TAGLIB_INTERN`: `MAKE_TAGLIB_LIB` is already defined and could be used.

**Comment by neheb** [comment]
@ufleisch the reason I added it is because compilation fails with -Werror otherwise.

**Comment by neheb** [comment]
Removed anyways.

File: taglib/toolkit/tfile.h [hunk 1]
```diff
@@ -66,17 +66,17 @@ namespace TagLib {
      * Specify which tags to strip either explicitly, or on save.
      */
     enum StripTags {
-      StripNone,  //<! Don't strip any tags
-      StripOthers //<! Strip all tags not explicitly referenced in method call
+      StripNone,  ///< Don't strip any tags
+      StripOthers ///< Strip all tags not explicitly referenced in method call
```
Comment (ufleisch) [inline taglib/toolkit/tfile.h:70]: `//!<` would be more consistent with the doxygen comment style used in the rest of the code.

File: taglib/toolkit/tfile.h [hunk 2]
```diff
@@ -66,17 +66,17 @@ namespace TagLib {
...
 
     /*!
      * Used to specify if when saving files, if values between different tag
      * types should be synchronized.
      */
     enum DuplicateTags {
-      Duplicate,     //<! Synchronize values between different tag types
-      DoNotDuplicate //<! Do not synchronize values between different tag types
+      Duplicate,     ///< Synchronize values between different tag types
+      DoNotDuplicate ///< Do not synchronize values between different tag types
```
Comment (ufleisch) [inline taglib/toolkit/tfile.h:79]: `//!<` would be more consistent with the doxygen comment style used in the rest of the code.

**Comment by neheb** [comment]
//<! is not valid Doxygen.

**Comment by ufleisch** [comment]
I am talking about `//!<`, not `//<!`.

**Pull request merged**